### PR TITLE
Introduce Container to manage grpcServer & listener lifecycle

### PR DESCRIPTION
**Pull request by nahguam**

### CODE REVIEW

File: server/container/container.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	c.log = log.With().
+		Str("container", name).
+		Str("bindAddress", listener.Addr().String()).
+		Logger()
+
+	go func() {
+		if err := c.server.Serve(listener); err != nil {
+			c.log.Fatal().Err(err).Msg("Failed to start serving")
+		}
+		fmt.Println("XXX")
```
Comment (andrasbeni) [inline server/container/container.go]:
```suggestion
```

File: server/container/container.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func Start(name string, port int, registerFunc func(grpc.ServiceRegistrar)) (*Container, error) {
+	c := &Container{
+		server: grpc.NewServer(),
+	}
+
+	registerFunc(c.server)
+
+	listener, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
```
Comment (andrasbeni) [inline server/container/container.go:26]: As I can see you are creating a new listener every time you call Start. But you call Start from both internal and public rpc server code, so there will be a conflict.

Comment (nahguam) [reply server/container/container.go:26]: Public and internal are configured to run on different ports, so there are 2 listeners.

Comment (nahguam) [reply server/container/container.go:26]: Would it be clearer if the function was `StartNew`?

Comment (andrasbeni) [reply server/container/container.go:26]: It was clear enough. What wasn't clear is that we use different ports.

**Pull request merged**